File: core/src/main/java/io/github/manami/core/services/events/RelativizeLocationEvent.java
package io.github.manami.core.services.events;

import io.github.manami.core.Manami;
import io.github.manami.core.commands.CmdChangeLocation;
import io.github.manami.dto.entities.Anime;

public class RelativizeLocationEvent extends AbstractEvent implements ReversibleCommandEvent {

  private final CmdChangeLocation command;


  public RelativizeLocationEvent(final Anime anime, final String newValue, final Manami app) {
    super(anime);
    command = new CmdChangeLocation(anime, newValue, app);
  }

  @Override
  public CmdChangeLocation getCommand() {
    return command;
  }
}

File: core/src/main/java/io/github/manami/core/services/events/RelativizeLocationEvent.kt
package io.github.manami.core.services.events

import io.github.manami.core.Manami
import io.github.manami.core.commands.CmdChangeLocation
import io.github.manami.dto.entities.Anime

class RelativizeLocationEvent(
        private val animeEntry: Anime,
        private val newValue: String,
        private val app: Manami
) : AbstractEvent(animeEntry), ReversibleCommandEvent {

    private val command: CmdChangeLocation = CmdChangeLocation(animeEntry, newValue, app)


    override fun getCommand(): CmdChangeLocation {
        return command
    }
}
